nilearn/mass_univariate/tests/test_permuted_least_squares.py

```diff
@@ -14,106 +16,217 @@
     all_kstest_pvals_intercept = []
     all_kstest_pvals_intercept2 = []
+
     # we compute the Mean Squared Error between cumulative Density Function
     # as a proof of consistency of the permutation algorithm
     all_mse = []
     all_mse_intercept = []
     all_mse_intercept2 = []
+
+    # test various number of permutations
+    perm_ranges = [10, 100, 1000]
+
     for i, n_perm in enumerate(np.repeat(perm_ranges, 10)):
         # Case no. 1: no intercept in the model
-        pval, orig_scores, h0 = permuted_ols(
-            tested_var,
-            target_var,
-            model_intercept=False,
-            n_perm=n_perm,
-            two_sided_test=False,
-            random_state=i,
-        )
-        assert_equal(h0.shape, (n_regressors, n_perm))
+        h0 = permuted_ols_no_intercept(tested_var, target_var, n_perm, i)
+        df = n_samples - 1
         h0_intercept = h0[0, :]
-
-        # Kolmogorov-Smirnov test
-        kstest_pval = stats.kstest(h0_intercept, stats.t(n_samples - 1).cdf)[1]
-        all_kstest_pvals.append(kstest_pval)
-        mse = np.mean(
-            (
-                stats.t(n_samples - 1).cdf(np.sort(h0_intercept))
-                - np.linspace(0, 1, h0_intercept.size + 1)[1:]
-            )
-            ** 2
-        )
-        all_mse.append(mse)
+        ks_stat_and_mse(df, h0_intercept, all_kstest_pvals, all_mse)
 
         # Case no. 2: intercept in the model
-        pval, orig_scores, h0 = permuted_ols(
-            tested_var,
-            target_var,
-            model_intercept=True,
-            n_perm=n_perm,
-            two_sided_test=False,
-            random_state=i,
-        )
-        assert_array_less(pval, 1.0)  # pval should not be significant
+        h0 = permuted_ols_with_intercept(tested_var, target_var, n_perm, i)
+        df = n_samples - 2
         h0_intercept = h0[0, :]
-
-        # Kolmogorov-Smirnov test
-        kstest_pval = stats.kstest(h0_intercept, stats.t(n_samples - 2).cdf)[1]
-        all_kstest_pvals_intercept.append(kstest_pval)
-        mse = np.mean(
-            (
-                stats.t(n_samples - 2).cdf(np.sort(h0_intercept))
-                - np.linspace(0, 1, h0_intercept.size + 1)[1:]
-            )
-            ** 2
+        ks_stat_and_mse(
+            df, h0_intercept, all_kstest_pvals_intercept, all_mse_intercept
         )
-        all_mse_intercept.append(mse)
 
         # Case no. 3: intercept in the model, no centering of tested vars
-        pval, orig_scores, h0 = permuted_ols(
+        permuted_ols_with_intercept(
             tested_var_not_centered,
             target_var,
-            model_intercept=True,
-            n_perm=n_perm,
-            two_sided_test=False,
-            random_state=i,
+            n_perm,
+            i,
         )
-        assert_array_less(pval, 1.0)  # pval should not be significant
+        df = n_samples - 2
         h0_intercept = h0[0, :]
-
-        # Kolmogorov-Smirnov test
```

Comment:
I think these are tested separately (case 1 and 2). Couldn't we split in multiple tests to make things clearer ?

Reply:
Let me actually try to split into 3 different tests